pkgs/development/python-modules/udatetime/default.nix

```diff
@@ -15,8 +15,9 @@ buildPythonPackage rec {
 
   patches = [
     # fix build with python 3.9
+    # https://github.com/freach/udatetime/33
     (fetchpatch {
-      url = "https://github.com/freach/udatetime/pull/33.patch";
+      url = "https://github.com/freach/udatetime/compare/75a07891426364f8bf0b44305b00bb1dd90534ae...2cfbc92cb274a80476a45c6c0d387c19e77a9f6e.patch";
       sha256 = "02wm7ivkv1viqn2wflgd10dgpddfqfrwacmrldigb1mwb79n554j";
```

Comment:
better set a name when doing a url like this because the filename will be weird

also

```
75a07891426364f8bf0b44305b00bb1dd90534ae...2cfbc92cb274a80476a45c6c0d387c19e77a9f6e.patch> 100  1450  100  1450    0     0   3947      0 --:--:-- --:--:-- --:--:--  3950
error: hash mismatch in fixed-output derivation '/nix/store/km14gn7xnadk5jpq3c9cxbdp2fk2p6ih-75a07891426364f8bf0b44305b00bb1dd90534ae...2cfbc92cb274a80476a45c6c0d387c19e77a9f6e.patch.drv':
         specified: sha256-kpRi01m8hvVio7kyxbPDrrX7GgjtUceFxXGHPXc8lQs=
            got:    sha256-pPskJnie+9H3qKqf8X37sxB+CH3lpkj7IYl8HfiuV/4=
```

Reply:
Old patch
```diff
--- a/src/rfc3339.c
+++ b/src/rfc3339.c
@@ -10,6 +10,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/timeb.h>
 #include <time.h>
```

new patch

```
From dbb04c9a427b61bc9df601492f426238550aa54f Mon Sep 17 00:00:00 2001
From: Stefano Rivera <stefano@rivera.za.net>
Date: Fri, 16 Oct 2020 21:19:00 -0700
Subject: [PATCH 1/2] Explicitly include <sys/timeb.h> for ftime() and struct
 timeb

Needed to build on Python 3.9.

Fixes: #32

Can't see what changed in the Python 3.9 headers, where this used to be
transitively included and isn't any more.

Note, ftime() is deprecated, and should be replaced with gettimeofday().
This PR does not do that.
---
 src/rfc3339.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/rfc3339.c b/src/rfc3339.c
index 1660028..126bbe8 100644
--- a/src/rfc3339.c
+++ b/src/rfc3339.c
@@ -10,6 +10,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/timeb.h>
 #include <time.h>
 
 

From 2cfbc92cb274a80476a45c6c0d387c19e77a9f6e Mon Sep 17 00:00:00 2001
From: Stefano Rivera <stefano@rivera.za.net>
Date: Thu, 24 Feb 2022 09:11:28 -0400
Subject: [PATCH 2/2] Wrap sys/timeb.h include in an ifdef HAVE_FTIME

---
 src/rfc3339.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/rfc3339.c b/src/rfc3339.c
index 126bbe8..e5fe1b0 100644
--- a/src/rfc3339.c
+++ b/src/rfc3339.c
@@ -10,9 +10,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/timeb.h>
 #include <time.h>
 
+#ifdef HAVE_FTIME
+#include <sys/timeb.h>
+#endif
 
 #define RFC3339_VERSION "0.0.16"
 #define DAY_IN_SECS 86400
```

Looks ok

Reply:
@globin

Reply:
[this one](https://github.com/freach/udatetime/pull/33/commits/dbb04c9a427b61bc9df601492f426238550aa54f.patch) works, but i'll update the hash